[PR] Adding concepts clarification and tutorial

LuchoTurtle opened the pull request:
closes #1 

This PR should be the initial template for the project. 
Has a section describing concepts for better understanding and a tutorial to follow through integration with Stripe.

nelsonic commented on README.md:
```diff
@@ -1,2 +1,363 @@
+There are some that can be acquired through
+a [one time purchase](https://git-fork.com/buy).
+There are others
+that are [subscription-based](https://www.notion.so/pricing).
```
Almost _all_ recent **`SaaS`** is subscription based.
We could have a _list_ of all the well-known **`SaaS`** products in a _footnote_.

we're _definitely_ not going to link to [`Notion`](https://user-images.githubusercontent.com/194400/85288902-75077480-b48e-11ea-8b41-75299936985b.png) in the intro paragraph to our payment processing tutorial. 🤣 
It's like FCP linking to [FC Barcelona](https://www.fcbarcelona.com/en/) on their home page ... 😜 
Not saying we will "compete" with `Notion`, but there will be some "overlap" and the _last_ thing we want is to send them _any_ more traffic. 👌

LuchoTurtle replied:
you're right
fixed on [`52e8e50` (#2)](https://github.com/dwyl/learn-payment-processing/pull/2/commits/52e8e50ea1dda267ef96ff6f8bd68941c1e1fce4)

LuchoTurtle commented:
Even though it's still in progress, @nelsonic  you can take a look at the "theory" part and check if there's anything big missing in it, before I start doing an example.
I didn't think going over the features of each alternative would be worthwhile from an engineering standpoint, as that is just marketing, imo.

LuchoTurtle commented:
Added Stripe integration. It should be fully working, with authentication and [`checkout sessions`](https://stripe.com/docs/payments/checkout).
I still want to expand the example more, though. Even though this works, I still want to block user access of unpaid users. Still need to see how that can be implemented with Stripe's API 🤔

nelsonic commented:
Sounds good. keep going. looking forward to reading your research findings. 👌

nelsonic commented on mix.exs:
```diff
@@ -0,0 +1,78 @@
+      {:phoenix_live_dashboard, "~> 0.7.2"},
+      {:esbuild, "~> 0.5", runtime: Mix.env() == :dev},
+      {:tailwind, "~> 0.1.8", runtime: Mix.env() == :dev},
+      {:swoosh, "~> 1.3"},
```
@LuchoTurtle please get into the habit of _excluding_ the stuff we _don't_ need from `Phoenix` projects. 🙏 
e.g: 

```sh
mix phx.new app --no-dashboard --no-ecto --no-gettext --no-mailer 
```

All the recent updates include `swoosh` for sending emails.
That's one more thing for us to receive annoying @depedabot updates for and have to _maintain_.
Please be ultra-precise with what you include in a project/tutorial. 
This example isn't going to need a fancy `Phoenix` dashboard, translation or even DB so it can be _significantly_ streamlined. Which will then make deploying the example to Fly.io _far_ more efficient (no Postgres instance required). 💭

LuchoTurtle replied:
You are completely right, I keep forgetting about this detail. Will do better next time.
We will be using a database, though. To keep track of the logged users who have paid or not paid.

nelsonic replied:
In a simple example I would be tempted to use **`ETS`** to store a JSON/Map of data: https://elixirschool.com/en/lessons/storage/ets
Running a separate VM for `Postgres` just to store **`2kb`** of data is like using a **Jet** to drive to the corner shop to buy bread. 😜

LuchoTurtle replied:
ETS is in-memory. Unless you want to lose data when a server restarts, I could give it a whirl.

nelsonic replied:
Yeah, ETS is often in-memory only, but can easily be persisted to disk: https://elixirforum.com/t/persistentets-lightweight-persistence-over-ets/3951

nelsonic commented on README.md:
```diff
@@ -1,2 +1,363 @@
+
+If we click the `Buy Fork` button,
+we are met with a modal 
+for a one-time purchase license.
```
This is not a good example. 
See: https://github.com/dwyl/product-ux-research/issues/38#issuecomment-1364001758
We do not want to promote this website because the UX is _horrible_.

A _much_ better but still not great example of a one-time payment for software is: https://eggerapps.at/postico2/
https://user-images.githubusercontent.com/194400/209484028-fa51dc93-7fd9-495a-9ab5-68682c955cfc.mov

Clear transition between screens and _single_ page checkout that clearly displays total price.

Created **`#research`** task on backlog to dive into this a bit deeper: https://github.com/dwyl/product-ux-research/issues/44

LuchoTurtle commented:
This should be done. I've added the project, used DETS for storage (no need for PostgreSQL here) and I got it running with 100% coverage.

Fixed some README typos and changed the `Fork` example you mentioned in the feedback.

I also added CI (but it didn't execute in this PR).

nelsonic commented:
![you-are-the-product](https://user-images.githubusercontent.com/194400/210129020-c3fe00a5-5721-448b-a919-7b9c89ba2fae.png)

nelsonic commented on README.md:
```diff
@@ -1,2 +1,1388 @@
+[That's a great question](https://www.indiehackers.com/post/stripe-vs-paddle-89161b0d5c).
+
+[Which seems to asked](https://splitbee.io/blog/why-we-moved-from-stripe-to-paddle)
+[over](https://stackshare.io/stackups/paddle-vs-stripe),
```
Please don't do this. `[over] and [over] and [over]` is not good link text. https://github.com/dwyl/contributing/issues/119
I _understand_ that you are emphasising the fact that _many_ people have had these same questions.
But as a document that we want to read, maintain and have others _contribute_ to. 
We want to set high standards for link text. 
Please/thanks! 🙏

nelsonic commented:
```sh
** (Mix) Expected one or more files/patterns to be given to mix format or for a .formatter.exs file to exist with an :inputs or :subdirectories key
```
https://github.com/dwyl/learn-payment-processing/actions/runs/3828017445/jobs/6513143646
<img width="819" alt="image" src="https://user-images.githubusercontent.com/194400/210326459-2d5cb827-a442-46e3-8df3-0cb96863cb72.png">

nelsonic commented:
https://github.com/dwyl/learn-payment-processing/actions/runs/3901637053/jobs/6663723897#step:6:152
![image](https://user-images.githubusercontent.com/194400/212057899-d1a738e7-8e02-4c83-be91-a26c8992ba69.png)

nelsonic reviewed (approved):
@LuchoTurtle this write-up is superb. 😍  Well done. 🎉

nelsonic commented:
Note: I made a few tweaks but your core content is excellent. Keep up the great work. 🚀
